frontend/packages/portal-frontend/src/compound/hooks/useRelatedCompoundsData.ts

```diff
@@ -0,0 +1,191 @@
+      if (!topTargets.has(correlatedCompoundGeneTarget)) {
+        topTargets.add(correlatedCompoundGeneTarget);
+      }
+    }
+
+    const correlatedCompoundName = correlatedCompound.other_dimension_label;
+    if (topCompounds.size < 10) {
+      if (!topCompounds.has(correlatedCompoundName)) {
+        topCompounds.add(correlatedCompoundName);
+      }
+    }
+    if (
+      topCompounds.has(correlatedCompoundName) &&
+      topTargets.has(correlatedCompoundGeneTarget)
+    ) {
+      filteredTopCorrelates.push(correlatedCompound);
+      const dataTypeKey = datasetToDataTypeMap[correlatedCompound.dataset];
+
+      if (!(correlatedCompoundName in topCorrelates)) {
+        topCorrelates[correlatedCompoundName] = {
+          CRISPR: {},
+          RNAi: {},
+        };
+      }
+      topCorrelates[correlatedCompoundName][dataTypeKey][
+        correlatedCompound.gene
+      ] = correlatedCompound.correlation;
+    }
+  }
+  return { topCorrelates, topTargets, topCompounds };
+}
+
+function useRelatedCompoundsData(
+  datasetId: string, // should be a compound dataset. Could be hardcoded instead of param..
+  compoundLabel: string,
+  datasetToDataTypeMap: Record<string, "CRISPR" | "RNAi">
+) {
+  const bapi = cached(breadboxAPI);
+
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [datasetName, setDatasetName] = useState<string>("");
+  const [targetCorrelationData, setTargetCorrelationData] = useState<
+    any | null
+  >(null);
+  const [topGeneTargets, setTopGeneTargets] = useState<string[]>([]);
+  const [topCompoundCorrelates, setTopCompoundCorrelates] = useState<string[]>(
+    []
+  );
+
+  useEffect(() => {
+    (async () => {
+      setIsLoading(true);
+
+      try {
+        // get compound dataset name
+        const compoundDataset = await bapi.getDataset(datasetId);
+        setDatasetName(compoundDataset.name);
+
+        // get gene targets for given compound
+        const compoundDimType = await bapi.getDimensionType("compound_v2");
+        if (compoundDimType.metadata_dataset_id) {
+          const allCompoundMetadata = await bapi.getTabularDatasetData(
+            compoundDimType.metadata_dataset_id,
+            {
+              identifier: "label",
+              columns: ["CompoundID", "GeneSymbolOfTargets"],
+            }
+          );
+          const targetGenes: string[] =
+            allCompoundMetadata.GeneSymbolOfTargets[compoundLabel] || [];
+
+          // make a list of datasets to correlate and gene target pairs to query by (ex: {dataset1, gene1}, {dataset2, gene1}, {dataset1, gene2}, {dataset2, gene2})
+          const datasetGenePairs = Object.keys(
+            datasetToDataTypeMap
+          ).flatMap((dataset) =>
+            targetGenes.map((gene) => ({ dataset, gene }))
+          );
+
+          // Here we are getting slices for each gene target in crispr and rnai datasets correlated with given compound dataset (Oncref)
```

Comment:
If OncRef is the only option, what happens for compounds that aren't available in the OncRef dataset?